qiskit/opflow/converters/circuit_sampler.py

```diff
@@ -198,7 +214,17 @@ def replace_circuits_with_dicts(operator, param_index=0):
             else:
                 return operator
 
-        if params:
+        # store the operator we constructed, if it isn't stored already
+        if op_id not in self._cached_ops.keys():
+            op_cache = OperatorCache()
+            op_cache.reduced_op_cache = self._reduced_op_cache
+            op_cache.circuit_ops_cache = self._circuit_ops_cache
+            op_cache.transpiled_circ_cache = self._transpiled_circ_cache
+            op_cache.transpile_before_bind = self._transpile_before_bind
+            op_cache.transpiled_circ_templates = self._transpiled_circ_templates
+            self._cached_ops[op_id] = op_cache
```

Comment:
Would we want a method to clear the cache at all? Otherwise I guess, if you are using the same CircuitSampler instance on different occasions, with different operators, this just continues to build up.

Reply:
We can add one, that's a good idea 👍🏻